subprojects/core/src/main/java/org/gradle/api/internal/tasks/TaskDestroyablesInternal.java

```diff
@@ -16,17 +16,12 @@
 
 package org.gradle.api.internal.tasks;
 
+import org.gradle.api.file.FileCollection;
 import org.gradle.api.tasks.TaskDestroyables;
 
-import java.io.File;
-import java.util.Collection;
-
 /**
  * Note: this is currently not visible on {@link org.gradle.api.internal.TaskInternal} to avoid it leaking onto {@link org.gradle.api.internal.AbstractTask} and so on to the public API.
  */
 public interface TaskDestroyablesInternal extends TaskDestroyables {
-    /**
-     * Returns the files that this task will destroy.
-     */
-    Collection<File> getFilesReadOnly();
+    FileCollection getFiles();
```

Comment:
Hmm...Maybe we shouldn't have this method match the removed method on the public interface.  If anyone _is_ using that method, it'll just continue to work, but they'll now be using an internal method and may not even know it.  Maybe change this to `getAllFiles()` or even just leave it as `getFilesReadOnly()`.  Then at least they'll get a breakage and if nothing else will complain about it and maybe surface a use case.

Reply:
Interestingly, I feel less of a problem about this than deprecating the `file()` and `files()` methods. I expect the number of people using `getFiles()` to be probably zero. OTOH, I feel going for `getAllFiles()` or `getFilesReadOnly()` instead of `getFiles()` is polluting the API, albeit only an internal one. But again, if you feel strongly about this, I'm happy to make the change.

Reply:
I don't feel strongly about this, either.  It was more an observation than anything else.  If you feel like changing the method name would pollute the api, I'm fine with it.